miner/worker.go

```diff
@@ -1116,11 +1152,18 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 			return
 		}
 		header.Coinbase = w.coinbase
+		if useMB {
+			header.Coinbase = maybeMB.Coinbase
+			header.Time = maybeMB.Timestamp
+			header.ParentHash = maybeMB.ParentHash
```

Comment:
This needs to be checked by the miner, and reject bundles that disagree.

Reply:
can you elaborate? Like timestamp is < current time, parent hash really is parent hash? I mean, in some sense, why even worry cause anyway it will just get rejected as invalid block anyway

Reply:
We don't want to allow the miner to produce blocks that will be immediately uncled

Reply:
fair - will make more defensive